### ISSUE TITLE
[RFE]: Support Storage pool options

### ISSUE THREAD

**Issue by aravindavk** [kadalu/kadalu#982]
Kadalu Storage pools are equivalent to Gluster Volumes. In external mode, the storage pool (Gluster volume or Kadalu pool) options are handled externally using Gluster command or Kadalu command. Add support for setting options for natively managed storage pools.

CRD changes:

- Introduce two new fields: `options`(object type) and `mount_options` (array type).

CLI changes:

- Storage add command accepts the Options and Mount options
- New command to set/reset Options and mount options (Ex: `kubectl kadalu set <POOL> ...`)
- Validate options

Operator changes:

- Validate the options and mount options
- Save in Config map(`handle_create`)
- In `handle_modified`:
    - Get existing config map info
    - Compare the options and mount options between request and existing(configmap).
    - If options changed,
        - Update configmap
        - Exec to respective server pods and regenerate volfiles
        - Notify(`SIGHUP`) `fsd` and `shd` processes.
- No support for handling mount options modification. It will be applied on next restart of CSI node plugin.

CSI changes:

- Use mount options from config map while mounting the pool (Options are already applied in Volfile, so no need to check).

Server changes:

- Use `options` from configmap while generating the volfiles.

**Note**: No support for setting options via StorageClass. Adding support here will make things complicated and makes it difficult to debug if the same storage pool behaves differently for some node plugin mounts. Also, it will be difficult to remount the storage pools with those options again on CSI restarts.

**Comment by leelavg** [comment]
> No support for setting options via StorageClass
- iirc, there was this feature also, so we want to stop that?
- I agree with the reasoning, just confirming the decision.

**Comment by vatsa287** [comment]
@leelavg Yes. I think most of that implementation is removed. If there are any leftovers will remove it too.

**Comment by aravindavk** [comment]
> iirc, there was this feature also, so we want to stop that?

Yes. Existing feature is not documented, and I think nobody using the same.
